src/renderer/dx/CustomTextLayout.cpp

```diff
@@ -63,6 +63,24 @@ CustomTextLayout::CustomTextLayout(gsl::not_null<IDWriteFactory1*> const factory
 
         _text += text;
     }
+
+    const auto textLength = gsl::narrow<UINT32>(_text.size());
+
+    BOOL isTextSimple = FALSE;
+    UINT32 uiLengthRead = 0;
+    const UINT32 glyphStart = 0;
+
+    _glyphIndices.resize(textLength);
+
+    const HRESULT hr = _analyzer->GetTextComplexity(
+        _text.c_str(),
+        textLength,
+        _font.Get(),
+        &isTextSimple,
+        &uiLengthRead,
+        &_glyphIndices.at(glyphStart));
+
+    _isEntireTextSimple = SUCCEEDED(hr) && isTextSimple && uiLengthRead == textLength;

```

Comment:
I figured that now there’s more than one place that used this property, then it should be ready as early as possible. Otherwise the value may not be initialized before being used.

Reply:
I'm just worried that it costs a lot on construct and that there's some roadmap where we recycle these layouts because the alloc/dealloc of the vectors is very expensive right now. But it can be moved around again later if you're reticent to move it again.

Reply:
I see. How about adding another method named “AnalyseTextComplexity” and put it before “AnalyseRuns”?

Reply:
That works for me.